[DOC] added minimal guidelines to organize tabular data (#696)
nipoppy/nipoppy: docs/source/how_to_guides/user_guide/organizing_tabular.md

Reviewer: This file should be moved outside of the legacy user guide. I think a good place would be either after the manifest guide or after "Running a pipeline"

This is the **legacy** user guide and may contain information that is out-of-date.
```

# Organizing tabular data

The question of managing tabular (e.g. clinical or behavioural) data is a complex one. The main challenges include

> 1. Lack of standardized vocabulary for naming variables

Reviewer: ```suggestion
1. Lack of a standardized vocabulary for naming variables
```

2. Asynchrony between tabular and imaging data collection workflows
3. Difficulty in defining criteria for data validation

> BIDS extension proposal ([BEP36](https://bids.neuroimaging.io/extensions/beps/bep_036.html)) is one of the major efforts in this space aimed at providing guidelines on standrded naming and organization of tabular data. Nipoppy will support and promote this standard once merged and therefore we do not have any strong rules or validations for tabular data management the moment. Having said that we do have a few recommendations that align with the BIDS direction and might be helpful in general.

Reviewer: Typo (somehow missed by Codespell...)

```suggestion
BIDS extension proposal ([BEP36](https://bids.neuroimaging.io/extensions/beps/bep_036.html)) is one of the major efforts in this space aimed at providing guidelines on standardized naming and organization of tabular data. Nipoppy will support and promote this standard once merged, and therefore we do not have any strong rules or validations for tabular data management the moment. Having said that, we do have a few recommendations that align with the BIDS direction and might be helpful in general.
```

## Source (i.e. acquired) data

> Similar to imaging data, it is good to separate "data collection" from "data curation" tasks even for the tabular data. This way we don't modify the acquired source data and only create "clean" curated copies. This is especially useful if you have different naming conventions for your `participant_ids` and/or `visit_ids` for the tabular vs imaging data. The recommended location for putting the "collected/acquired" data dump is `<dataset_root>/sourcedata/tabular` directory.

Reviewer: Minor formatting things + use substitution for path

```suggestion
Similar to imaging data, it is good to separate "data collection" from "data curation" tasks even for the tabular data. This way we don't modify the acquired source data and only create "clean" curated copies. This is especially useful if you have different naming conventions for your `participant_id`s and/or `visit_id`s for the tabular vs imaging data. The recommended location for putting the "collected/acquired" data dump is {{dpath_src_tabular}} directory.
```

```{note}

> If you do have different naming conventions for the clinical visits vs imaging sessions, then you can establish the correct mapping between those (e.g. V01 <-> ses-BL) in the [manifest](../manifest/index) file.

Reviewer: ```suggestion
If you do have different naming conventions for the clinical visits vs imaging sessions, then you can establish the correct mapping between those (e.g. `V01` <-> `ses-BL`) in the [manifest](../manifest/index) file.
```

```

## Demographic variables

> You can then write custom scripts to generate "curated" data files from source data dump. These files will go in the `<dataset_root>/tabular` directory. Here we usually recommend first creating a "demographics.tsv" file that includes typical demographic variables such as dob/age, sex, recruitment_cohort, screening_date etc. collected by most studies. One can also think of this file as the basic participant info recorded at recruitment / screening visit that is static and does not change over time. However, since we don't validate any tabular files, you can include multiple visits per participants here if preferred.

Reviewer: Minor rewording

```suggestion
You can then write custom scripts to generate "curated" data files from source data dump. These files will go in the {{dpath_tabular}} directory. Here we usually recommend first creating a `demographics.tsv` file that includes typical demographic variables collected by most studies, such as date of birth/age, sex, recruitment cohort, screening date etc. One can also think of this file as the basic participant information recorded at a recruitment / screening visit that is static and does not change over time. However, since we don't validate any tabular files, you can include multiple visits per participants here if preferred.
```

### Example demographics TSV file

:::{csv-table}
---
file: ./inserts/example1-demographics.tsv
header-rows: 1
delim: tab
---
:::

## Behavioural and clinical data

> For the behavioural or clinical assessments, we create a `<dataset_root>/tabular/assessment` directory and then generate single TSV file (e.g. assessment_A.tsv) per assessment/instrument. This file contains separate row per `participant_id` and `visit_id` (or `session_id` if identical). This modularity at the level of assessment is helpful for quality checks and making corrections or updates. This file organiation is also **not** validated by Nippopy, so one can come up alternative ways to organize / split clinical assessment information into separate files as preferred.

Reviewer: You can update the MyST substitution dictionary in conf.py:
```python
    "dpath_assessments": f"`{DEFAULT_LAYOUT_INFO.dpath_assessments}`",
```

Then use `{{dpath_assessments}}` in the Markdown for substitution

> For the behavioural or clinical assessments, we create a `<dataset_root>/tabular/assessment` directory and then generate single TSV file (e.g. assessment_A.tsv) per assessment/instrument. This file contains separate row per `participant_id` and `visit_id` (or `session_id` if identical). This modularity at the level of assessment is helpful for quality checks and making corrections or updates. This file organiation is also **not** validated by Nippopy, so one can come up alternative ways to organize / split clinical assessment information into separate files as preferred.

Reviewer: ```suggestion
For the behavioural or clinical assessments, we create a `<dataset_root>/tabular/assessment` directory and then generate single TSV file (e.g. `assessment_A.tsv`) per assessment/instrument. This file contains separate row per `participant_id` and `visit_id` (or `session_id` if identical). This modularity at the level of assessment is helpful for quality checks and making corrections or updates. This file organization is also **not** validated by Nippopy, so one can come up alternative ways to organize / split clinical assessment information into separate files as preferred.
```

### Example assessment TSV file

:::{csv-table}
---
file: ./inserts/example1-assessment.tsv
header-rows: 1
delim: tab
---
:::

### Data dictionaries

> For each file generated, it is recommended to create data dictionary i.e. demographics.json, assessment_A.json etc. next to the data file itself (see examples in [BIDS docs](https://bids.neuroimaging.io/getting_started/folders_and_files/metadata/json.html)). Nipoppy or BIDS itself doesn't help with creating "standardized" data dictionaries, but another related project [Neurobagel](https://annotate.neurobagel.org/) can help you with it. Neurobagel provides a simple annotation tool to help generate the data dictionaries with standardized vocabulary (when available) for your variable names. This will allow you to harmonize variable naming across different datasets (e.g. all demographic variables will be mapped to the same term) - which can be super helpful when you want to combine, compare or analyze multiple datasets.

Reviewer: ```suggestion
For each file generated, it is recommended to create data dictionary i.e. `demographics.json`, `assessment_A.json` etc. next to the data file itself (see examples in [BIDS docs](https://bids.neuroimaging.io/getting_started/folders_and_files/metadata/json.html)). Nipoppy or BIDS itself doesn't help with creating "standardized" data dictionaries, but another related project [Neurobagel](https://annotate.neurobagel.org/) can help you with it. Neurobagel provides a simple annotation tool to help generate the data dictionaries with standardized vocabulary (when available) for your variable names. This will allow you to harmonize variable naming across different datasets (e.g. all demographic variables will be mapped to the same term) - which can be super helpful when you want to combine, compare or analyze multiple datasets.
```
